src/Generator.Bind/GL2/Generator.cs

```diff
@@ -120,11 +120,33 @@ public virtual void Process()
                 SpecReader.ReadDelegates(file, Delegates, Profile, Version);
             }
 
+            HashSet<string> extensions = new HashSet<string>();
+            foreach (var (name, list) in Delegates)
+            {
+                var @delegate = list[0];
+                if (string.IsNullOrEmpty(@delegate.Category) == false)
+                {
+                    foreach (var part in @delegate.Category.Split('|'))
+                    {
+                        extensions.Add(part);
+                    }
+                }
+            }
+
+            foreach(var (name, @enum) in Enums)
+            {
+                var match = Utilities.Extensions.Match(name);
+                if (match.Index == 0 && match.Length != 0)
+                {
+                    extensions.Add(name);
+                }
+            }
+
```

Comment:
Writing this section as Linq would likely make it a lot more obvious that these loops are doing nothing more than populating the HashSet with a projection of the original data:
```cs
extensions.UnionWith(Delegates
    .SelectMany(d => (d.Item2[0].Category ?? string.Empty).Split('|'))
    .Where(s => !string.IsNullOrEmpty(s));

Match match;
extensions.UnionWith(Enums
    .Select(e => e.Item1)
    .Where(s => (match = Utilities.Extensions.Match(s)).Index == 0 && match.Length != 0));
```
By writing it as Linq, it's short, simple, and a lot more obvious that it's just projecting and filtering, and importantly, it's more obvious that there are no side effects in the loops.

Reply:
Making this into linq would not improve readability in my mind. I will keep it like it is.

Reply:
I understand your concern, but I do not share it. This code is simple to look at, simple to verify.
The code is already full of mutable state that changes in much more cursed ways than potentially allowing this one hash set to be modified.